src/cdk/overlay/dispatchers/overlay-click-dispatcher.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {DOCUMENT} from '@angular/common';
+import {Inject, Injectable} from '@angular/core';
+import {OverlayRef} from '../overlay-ref';
+import {Platform} from '@angular/cdk/platform';
+import {OverlayBaseDispatcher} from './overlay-base-dispatcher';
+
+/**
+ * Service for dispatching mouse click events that land on the body to appropriate overlay ref,
+ * if any. It maintains a list of attached overlays to determine best suited overlay based
+ * on event target and order of overlay opens.
+ */
+@Injectable({ providedIn: 'root' })
+export class OverlayClickDispatcher extends OverlayBaseDispatcher {
+  private _cursorOriginalValue: string;
+  private _cursorStyleIsSet = false;
+
+  constructor(@Inject(DOCUMENT) document: any, private _platform: Platform) {
+    super(document);
+  }
+
+  /** Add a new overlay to the list of attached overlay refs. */
+  add(overlayRef: OverlayRef): void {
+    super.add(overlayRef);
+
+    // Lazily start dispatcher once first overlay is added
+    if (!this._isAttached) {
+      this._document.body.addEventListener('click', this._clickListener, true);
+
+      // click event is not fired on iOS. To make element "clickable" we are
+      // setting the cursor to pointer
+      if (this._platform.IOS && !this._cursorStyleIsSet) {
+        this._cursorOriginalValue = this._document.body.style.cursor;
+        this._document.body.style.cursor = 'pointer';
+        this._cursorStyleIsSet = true;
+      }
+
+      this._isAttached = true;
+    }
+  }
+
+  /** Detaches the global keyboard event listener. */
+  protected detach() {
+    if (this._isAttached) {
+      this._document.body.removeEventListener('click', this._clickListener, true);
+      if (this._platform.IOS && this._cursorStyleIsSet) {
+        this._document.body.style.cursor = this._cursorOriginalValue;
+        this._cursorStyleIsSet = false;
+      }
+      this._isAttached = false;
+    }
+  }
+
+  /** Click event listener that will be attached to the body propagate phase. */
+  private _clickListener = (event: MouseEvent) => {
+    const overlays = this._attachedOverlays;
+
+    // Dispatch the mouse event to the top overlay which has subscribers to its mouse events.
+    // We want to target all overlays for which the click could be considered as outside click.
+    // As soon as we reach an overlay for which the click is not outside click we break off
+    // the loop.
+    for (let i = overlays.length - 1; i > -1; i--) {
+      const overlayRef: OverlayRef = overlays[i];
+      if (overlayRef._clickEventSubscriptions < 1) {
+        continue;
+      }
+
+      const config = overlayRef.getConfig();
+      const elements = [...config.excludeFromOutsideClick!];
+      elements.push(overlayRef.overlayElement);
```

Comment:
I would name this `excludedElements`. You can also omit the push and include the element in the array directly:
```ts
const excludedElements = [...config.excludeFromOutsideClick, overlayRef.overlayElement];
```